pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -219,24 +221,31 @@ func reportMetrics(pa *pav1alpha1.PodAutoscaler, pc podCounts) error {
 }
 
 // computeActiveCondition updates the status of a PA given the current scale (got), desired scale (want)
-// and the current status, as per the following table:
+// active threshold (min), and the current status, as per the following table:
 //
-//    | Want | Got    | Status     | New status |
-//    | 0    | <any>  | <any>      | inactive   |
-//    | >0   | < min  | <any>      | activating |
-//    | >0   | >= min | <any>      | active     |
-//    | -1   | < min  | inactive   | inactive   |
-//    | -1   | < min  | activating | activating |
-//    | -1   | < min  | active     | activating |
-//    | -1   | >= min | inactive   | inactive   |
-//    | -1   | >= min | activating | active     |
-//    | -1   | >= min | active     | active     |
+//    | Want | Got    | min   | Status     | New status |
+//    | 0    | <any>  | <any> | <any>      | inactive   |
+//    | >0   | < min  | <any> | <any>      | activating |
+//    | >0   | >= min | <any> | <any>      | active     |
+//    | -1   | < min  | <any> | inactive   | inactive   |
+//    | -1   | < min  | <any> | activating | activating |
+//    | -1   | < min  | <any> | active     | activating |
+//    | -1   | >= min | <any> | inactive   | inactive   |
+//    | -1   | >= min | 0     | activating | inactive   |
+//    | -1   | >= min | 0     | active     | inactive   | <-- this case technically is impossible.
+//    | -1   | >= min | >0    | activating | active     |
+//    | -1   | >= min | >0    | active     | active     |
 func computeActiveCondition(ctx context.Context, pa *pav1alpha1.PodAutoscaler, pc podCounts) {
 	minReady := activeThreshold(ctx, pa)
+	if pc.ready >= minReady && (pa.Status.IsSKSReady() ||
+		// In the initial scale 0 case, there won't be any endpoints ready, and therefore SKS will still be not ready.
+		(!pa.Status.IsSKSReady() && pa.Status.GetCondition(pav1alpha1.PodAutoscalerSKSReady).Message != noPrivateServiceName)) {
+		pa.Status.MarkScaleTargetInitialized()
+	}
 
 	switch {
-	case pc.want == 0:
-		if pa.Status.IsActivating() {
+	case pc.want == 0 || minReady == 0:
```

Comment:
do we need to do `|| minReady==0`? It seems that pc.want=0 is a superset of that?  E.g. if minReady=0, pc.want _will be_ 0 (the opposite is not true, though).

Reply:
We aren't overriding -1 with 0, so we are still hitting the pc.want = -1, minReady = 0 case.

Reply:
hm, I had that state somewhere in my head, but I guess we got rid of that :-)

Reply:
yeah originally we were overriding -1 with 0, but with Markus' simplification we are able to get rid of it.